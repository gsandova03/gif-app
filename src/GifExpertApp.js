import React, { useState } from 'react'
import { AddCategory } from './componentes/AddCategory';
import { SearchCategory } from './componentes/SearchCategory';

const GifExpertApp = ({ defaultCategories = [] }) => {

  // const [categories, setCategories] = useState(['One Punch']);
  const [categories, setCategories] = useState( defaultCategories );

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr/>

      <ul>
        {
          categories.map( ( category ) => (
            <SearchCategory key={ category } category={category} />
          ))
        }
      </ul>
    </>
  )
}

export default GifExpertApp;
