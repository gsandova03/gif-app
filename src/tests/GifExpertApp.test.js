import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'

describe('Pruebas en el componente <GifExpertApp />' , ()=> {

  test('debe de mostrar el componente correctamente', () => {
    
    const wrapper = shallow( <GifExpertApp /> )

    expect( wrapper ).toMatchSnapshot();

  })

  test('debe de mostar una lista de categorias', () => {
    const categories = ['One Punch', 'Hunter X Hunter'];

    const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> )

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('SearchCategory').length ).toBe( categories.length );
  })
  
})