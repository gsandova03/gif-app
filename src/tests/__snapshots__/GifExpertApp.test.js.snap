// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> debe de mostar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <SearchCategory
      category="One Punch"
      key="One Punch"
    />
    <SearchCategory
      category="Hunter X Hunter"
      key="Hunter X Hunter"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
